src/App.tsx

```diff
@@ -1,24 +1,217 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useState,
+  useEffect,
+  useMemo,
+  useCallback,
+} from 'react';
 
-const USER_ID = 0;
+import { TodoList } from './components/TodoList';
+import { TodoHeader } from './components/TodoHeader';
+import { ErrorNotification } from './components/ErrorNotification';
+
+import {
+  getTodos,
+  createTodo,
+  deleteTodo,
+  updateTodoStatus,
+  updateTodoTitle,
+} from './api/todos';
+
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
+import { TodoFooter } from './components/TodoFooter';
+
+import { getVisibleTodos } from './utils/helpres';
+
+const USER_ID = 10681;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [filter, setFilter] = useState(TodoStatus.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodos, setLoadingTodos] = useState([0]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((fetchedTodos: Todo[]) => {
+        setTodos(fetchedTodos);
+      })
+      .catch((fetchedError: Error) => {
+        setError(fetchedError?.message ?? 'Something went wrong');
+      });
+  }, []);
+
+  useEffect(() => {
+    let timeout: NodeJS.Timeout;
+
+    if (error) {
+      timeout = setTimeout(() => {
+        setError(null);
+      }, 3000);
+    }
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [error]);
+
+  const activeTodos = useMemo(() => (
+    todos.filter(todo => !todo.completed)
+  ), [todos]);
+
+  const completedTodos = useMemo(() => (
+    todos.filter(todo => todo.completed)
+  ), [todos]);
+
+  const handleAddTodo = useCallback(async (todoTitle: string) => {
+    try {
+      const newTodo = {
+        userId: USER_ID,
+        title: todoTitle.trim(),
+        completed: false,
+      };
+
+      setTempTodo({
+        id: 0,
+        ...newTodo,
+      });
+
+      const createdTodo = await createTodo(newTodo);
+
+      setTodos(currentTodos => [...currentTodos, createdTodo]);
+    } catch {
+      setError('Unable to add a todo');
+    } finally {
+      setTempTodo(null);
+    }
```

Comment:
here you use try/catch/finally with async/await syntax, above you use chaining with then + catch, why don't you use the same approach everywhere? I suggest rewriting evrerything with modern async/await syntax